### PR TITLE
exit with error if no tests are found

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton ptal.

File: hack/test-integration.sh [hunk 1]
```diff
@@ -60,5 +60,8 @@ export childargs
 
 # run each test as its own process
 pushd "./${package}" 2>&1 >/dev/null
-time go run "${OS_ROOT}/hack/listtests.go" -prefix="${OS_GO_PACKAGE}/${package}.Test" "${testdir}" | xargs -I {} -n 1 bash -c "exectest {} ${@:1}" # "${testexec}" -test.run="^{}$" "${@:1}"
+# first run just lists the tests and ensures the list is non-zero length.  listtests will return non-zero
+# if no tests are found.
+go run "${OS_ROOT}/hack/listtests.go" -prefix="${OS_GO_PACKAGE}/${package}.Test" "${testdir}" > /dev/null
```
Comment (smarterclayton) [inline hack/test-integration.sh]: Shouldn't pipefail catch this?

Comment (bparees) [inline hack/test-integration.sh]: it didn't when i tried returning non-zero from listgo, but maybe log.Fatalf does something more severe? i'll test it.

Comment (smarterclayton) [inline hack/test-integration.sh]: All fatalf should exit code 1, which *should* trigger pipefail.

**Comment by bparees** [comment]
@smarterclayton evidently yes.  updated.

File: hack/test-integration.sh (hunk 1 above)
Comment (bparees) [inline hack/test-integration.sh]: well what's odd is i first tried os.Exit(1) and that did not trigger pipefail i don't think.

in hindsight maybe my test wasn't very good though, so maybe that would have worked.  oh well, log.Fatalf is a better solution anyway.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**